lib/spaceship/tunes/build_train.rb

```diff
@@ -88,17 +88,25 @@ def update_testing_status!(new_value, testing_type, build = nil)
 
         build ||= latest_build if testing_type == 'external'
 
-        data['trains'].each do |train|
-          train["#{testing_type}Testing"]['value'] = false
-          train["#{testing_type}Testing"]['value'] = new_value if train['versionString'] == version_string
-
-          # find correct build
-          train['builds'].select! { |b| !b["#{testing_type}Testing"].nil? && !build.nil? && b['buildVersion'] == build.build_version }
-
-          # also update the build if it was found
-          train['builds'].each do |b|
-            b["#{testing_type}Testing"]['value'] = false
-            b["#{testing_type}Testing"]['value'] = new_value if b['trainVersion'] == version_string
+        # Delete the irrelevant trains and update the relevant one to enable testing
+        data['trains'].delete_if do |train|
+          if train['versionString'] != version_string
```

Comment:
Sorry I was mistaken, the delete_if here is definitely required so that only the trains of interest are updated.